handlers/SolanaChainHandler.ts

```
@@ -1,82 +1,313 @@
+    const finalizedDeposit = await super.finalizeDeposit(deposit);
+  
+    if (!finalizedDeposit?.receipt) {
+      return;
+    }
+    LogMessage(`Finalizing deposit ${deposit.id} on Solana...`);
+  
+    const l1Receipt = finalizedDeposit.receipt;
+    if (!l1Receipt) {
+      LogWarning(
+        `No finalize receipt found for deposit ${deposit.id}; cannot parse logs.`
+      );
+      return;
+    }
+    
+    let transferSequence: string | null = null;
+    try {
+      const logs = l1Receipt.logs || [];
+  
+      for (const log of logs) {
+        if (log.topics[0] === TOKENS_TRANSFERRED_SIG) {
+          const parsedLog = this.l1BitcoinDepositor.interface.parseLog(log);
+          const { transferSequence: seq } = parsedLog.args;
+          transferSequence = seq.toString();
+          break;
+        }
+      }
+    } catch (error: any) {
+      LogError(`Error parsing L1 logs for deposit ${deposit.id}`, error);
+    }
+  
+    if (!transferSequence) {
+      LogWarning(
+        `Could not find transferSequence in logs for deposit ${deposit.id}.`
+      );
+      return;
+    }
+
+    await updateToAwaitingWormholeVAA(l1Receipt.transactionHash, deposit, transferSequence);
+    LogMessage(`Deposit ${deposit.id} now awaiting Wormhole VAA.`);
+  }
+
+  public async bridgeSolanaDeposit(deposit: Deposit): Promise<void> {
+    if (!this.connection || !this.provider || !this.wallet) {
+      // Solana connection not initialized
+      LogWarning(`Solana connection not initialized. Cannot bridge deposit ${deposit.id}.`);
+      return;
+    }
+
+    if (!this.wormholeGatewayProgram) {
+      LogWarning(`Wormhole Gateway program not initialized. Cannot bridge deposit ${deposit.id}.`);
+      return;
+    }
+
+    if (deposit.status !== DepositStatus.AWAITING_WORMHOLE_VAA) return;
+
+    const secretKeyBase64 = this.config.solanaSignerKeyBase;
+    if (!secretKeyBase64) throw new Error('Missing solanaSignerKeyBase');
```

Not related to this PR, but thinking ahead about the future of this codebase: For all services, we may want to check their dependencies and fail early during initialization